intentv1/intent.ridl

```diff
@@ -0,0 +1,103 @@
+webrpc = v1
+
+name = sequence-waas-intents
+version = v0.1.0
+
+struct Signature
+  - sessionId: string
+  - signature: string
+
+# no way to generate string enums
+# enum IntentName: uint8
+#  - openSession
+#  - closeSession
+#  - validateSession
+#  - finishValidateSession
+#  - listSessions
+#  - getSession
+#  - sign
+#  - transaction
+
+struct Intent
+  - version: string
+  - name: string
+  - expires: uint64
+  - issued: uint64
+  - data: any
+  - signatures: []Signature
+
+struct SessionPacketProof
+  - email?: string
+  - idToken?: string
+
+struct IntentDataOpenSession
+  - sessionId: string
+  - proof: SessionPacketProof
+
+struct IntentDataCloseSession
+  - sessionId: string
+
+struct IntentDataValidateSession
+  - sessionId: string
+  - deviceMetadata: string
+
+struct IntentDataFinishValidateSession
+  - sessionId: string
+  - salt: string
+  - challenge: string
+
+struct IntentDataListSessions
+
+struct IntentDataGetSession
+  - sessionId: string
+
+struct IntentDataSign
+  - network: string
+  - message: string
+
+struct IntentDataTransaction
```

Comment:
I think this should be "SendTransaction" instead of just "Transaction", as its more clear. But yes lets discuss the prefix still etc,... as "IntentDataXXX" doesn't feel quiet right yet